client/src/core/client/framework/lib/rest.ts

```diff
@@ -45,6 +44,10 @@ const handleResp = async (res: Response) => {
     return res.text();
   }
 
+  if (type && type.includes("application/octet-stream")) {
+    return res.arrayBuffer();
+  }
+
```

Comment:
what was the reason for this change? I'm sure it's necessary, but I'm at a loss as to what it's doing because this whole prior written handler is somewhat obscure.

Reply:
This is to support the CSV download for DSA reports from the API. Otherwise, it won't work since it tries to respond with json and errors.